core/src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -1,10 +1,27 @@
 package pureconfig
 
+/**
+ * A mapping between case class fields and their respective keys in the config.
+ */
 trait ConfigFieldMapping extends (String => String) {
+
   def apply(fieldName: String): String
+
+  /**
+   * Returns a `ConfigFieldMapping` that uses this mapping with some exceptions.
+   *
+   * @param exceptions the exceptions for this mapping as pairs (field, configKey)
```

Comment:
@leifwickland I understand the confusion, in older pureconfig versions `ConfigFieldMapping` was the object to provide implicitly to configure product derivation :) Since `ProductHint`s were added, `ConfigFieldMapping` was made only one of the multiple things you can configure via a `ProductHint`.

@derekmorr yeah, I wonder about that too. I don't think that the current name is perfect, but I'm not sure of `withOverrides` is less explicit/correct in the context of a mapping... I ended up using `Exceptions`. @leifwickland, @melrief, @jcazevedo, what do you think?